fastchat/serve/model_worker.py

```diff
@@ -243,20 +244,47 @@ def generate_gate(self, params):
     def get_embeddings(self, params):
         try:
             tokenizer = self.tokenizer
-            input_ids = tokenizer.encode(params["input"], return_tensors="pt").to(
-                self.device
-            )
-            model_output = self.model(input_ids, output_hidden_states=True)
-            is_chatglm = "chatglm" in str(type(self.model)).lower()
-            if is_chatglm:
-                data = (model_output.hidden_states[-1].transpose(0, 1))[0]
+            is_vicuna = "llama" in str(type(self.model)) # vicuna support batch inference
+            is_chatglm = "chatglm" in str(type(self.model))
+            if is_vicuna:
+                encoding = tokenizer.batch_encode_plus(
+                    params["input"], padding=True, return_tensors="pt"
+                )
+                input_ids = encoding["input_ids"].to(self.device)
+                attention_mask = encoding["attention_mask"].to(self.device)
+                model_output = self.model(
+                    input_ids, attention_mask, output_hidden_states=True
+                )
+                data = model_output.hidden_states[-1]
```

Comment:
use `last_hidden_state`

Reply:
I tried `last_hidden_state` here, but an error occurred: 
`AttributeError: 'CausalLMOutputWithPast' object has no attribute 'last_hidden_state`
For  CausalLMOutput, there is no `last_hidden_state` member: [reference](https://huggingface.co/docs/transformers/main_classes/output#transformers.modeling_outputs.CausalLMOutput.hidden_states)